docs/schedule_scripting_language.md

```diff
@@ -0,0 +1,901 @@
+
+If the condition of any of these commands is not satisfied and said command have
+an `else`, then said the control flow jumps to the body of that `else`. If
+there's no `else` and the condition isn't satisfied then the control flow just
+fallthroughs into the next command.
+
+#### `if_week_event_reg`
+
+Checks if the passed WeekEventReg flag is set, and execute the body of the
+command if it is set.
+
+##### `if_week_event_reg` arguments
+
+- Argument 0: A WeekEventReg flag. `WEEKEVENTREG_` macros are preferred.
+
+##### `if_week_event_reg` example
+
+```c
+if_week_event_reg (WEEKEVENTREG_61_02) {
+    return_s (31)
+}
+return_s (30)
+```
+
+##### `if_week_event_reg` non-generics
+
+`if_week_event_reg_s` and `if_week_event_reg_l`
+
+#### `if_time_range`
+
+Checks if the current time is between the passed time range.
+
+##### `if_time_range` arguments
+
+- Argument 0: Hour component of the start time
+- Argument 1: Minute component of the start time
+- Argument 2: Hour component of the end time
+- Argument 3: Minute component of the end time
+
+##### `if_time_range` example
+
+```c
+// Checks if the current time is between 18:00 ~ 6:00
+if_time_range (18, 0, 6, 0) {
+    return_time (18, 0, 6, 0, 2)
+} else {
+    return_none
+}
+```
+
+##### `if_time_range` non-generics
+
+`if_time_range_s` and `if_time_range_l`
+
+#### `if_misc`
+
+Checks if the passed miscellaneous argument is true.
+
+##### `if_misc` arguments
+
+- Argument 0: A value of the `ScheduleCheckMisc` enum.
+
+##### `if_misc` example
+
+```c
+if_misc (SCHEDULE_CHECK_MISC_MASK_ROMANI) {
+    return_s (33)
+} else {
+    return_s (34)
+}
+```
+
+##### `if_misc` non-generics
+
+`if_misc_s`. Note there's no long version of this command. It is advised to
+minimize the amount of commands used on a `ìf_misc` body and `else`'s body.
+
+#### `if_scene`
+
+Checks if the current scene matches the one passed as parameter.
+
+##### `if_scene` arguments
+
+- Argument 0: A value of the `SceneId` enum.
+
+##### `if_scene` example
+
+```c
+if_scene (SCENE_SECOM) {
+    return_s (7)
+}
+```
+
+##### `if_scene` non-generics
+
+`if_scene_s` and `if_scene_l`
+
+#### `if_day`
+
+Checks if the current day matches the passed argument.
+
+##### `if_day` arguments
+
+- Argument 0: The day to check.
+
+##### `if_day` example
+
+```c
+if_day (3) {
+    return_l (1)
+}
+```
+
+##### `if_day` non-generics
+
+`if_day_s` and `if_day_l`
+
+#### `if_before_time`
+
+Checks if the current time is before the time passed as parameter.
+
+##### `if_before_time` arguments
+
+- Argument 0: The hour component of the time.
+- Argument 1: The minute component of the time.
+
+##### `if_before_time` example
+
+```c
+if_before_time (8, 0) {
+    return_s (19)
+} else {
+    return_none
+}
+```
+
+##### `if_before_time` non-generics
+
+`if_before_time_s` and `if_before_time_l`
+
+##### `if_before_time` notes
+
+This command performs the contrary check to `if_since_time`.
+
+#### `if_since_time`
```

Comment:
To me "after" sounds like a `>` check, while "since" sounds like a `>=` check.
At least to me it felt like an important distinction to make since this command is just syntactic sugar for `not if_before_time` (like, it isn't a command that maps exactly to a command from `z_schedule.c`).
If this difference isn't worth the trouble then we could just go with `if_after_time`